### PR TITLE
ND electrons computeBulkVelocity

### PR DESCRIPTION
**Pull request by Dekken**
the changes ndarray_vector.h may be unacceptable, I thought it was useful to remove "if constexpr dim == "

this was a guess: 
https://github.com/PHAREHUB/PHARE/pull/230/files#diff-5bc7e20169f26980f29061f3d35cb9d4R99

### CODE REVIEW

**Comment by Dekken** [comment]
> can be merged once the lower/upper XYZ are not taken form XYZ components of Ve

even with the ndarray changes?

**Comment by nicolasaunai** [comment]
> > can be merged once the lower/upper XYZ are not taken form XYZ components of Ve
> 
> even with the ndarray changes?

well... what's the alternative ? I see two : 

- back to if constexpr...
- get rid of the 3 int operator() in field 1D and 2D and make operator(Index<dim>(ix,iy,iz)) that does basically what your sub_array does, i.e. makes an index ignoring all above dim-1.
- something else?

**Comment by Dekken** [comment]
I'll see about #2

File: src/core/data/ndarray/ndarray_vector.h [hunk 1]
```diff
@@ -134,12 +146,7 @@ namespace core
     template<std::size_t dim>
     auto makeNdArray(std::array<std::uint32_t, dim> sizes)
     {
-        if constexpr (dim == 1)
-            return NdArrayVector<1>{sizes[0]};
-        if constexpr (dim == 2)
-            return NdArrayVector<2>{sizes[0], sizes[1]};
-        if constexpr (dim == 3)
-            return NdArrayVector<3>{sizes[0], sizes[1], sizes[2]};
+        return NdArrayVector<dim>{sizes};
```
Comment (Dekken) [inline src/core/data/ndarray/ndarray_vector.h:149]: this function (makeNdArray) seems kinda redundant now

Comment (nicolasaunai) [reply src/core/data/ndarray/ndarray_vector.h:149]: did you try to delete it?

Comment (Dekken) [reply src/core/data/ndarray/ndarray_vector.h:149]: nope but I will I think

**Comment by Dekken** [comment]
regarding the commit message "might want to profile this"

I believe was running with "-O0 -g3", whereas "-O3 -g0" runs in much less time (as expected)

File: src/core/data/ndarray/ndarray_vector.h (hunk 1 above)
Comment (Dekken) [reply src/core/data/ndarray/ndarray_vector.h]: removed

File: tests/simulator/refined_particle_nbr.py [hunk 2]
```diff
@@ -119,9 +119,9 @@ def test_1d(self):
     PREVIOUS_ITERATION_MAX_DIFF_2d = 1.50
 
     ## blocked by "electrons.h computeBulkVelocity()" - not updated for 2d
```
Comment (nicolasaunai) [inline tests/simulator/refined_particle_nbr.py:121]: remove?

File: tools/bench/sim/split.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python3
+#
+# formatted with black
+import os
+os.environ["OMP_NUM_THREADS"] = "1" # export OMP_NUM_THREADS=1
```
Comment (nicolasaunai) [inline tools/bench/sim/split.py:5]: could you comment a bit on why we have to set these env var?

Comment (Dekken) [reply tools/bench/sim/split.py:9]: done

File: tests/simulator/refined_particle_nbr.py (hunk 2 above)
Comment (Dekken) [reply tests/simulator/refined_particle_nbr.py]: dpme

**Comment by nicolasaunai** [comment]
question : do we keep these 8 commits into master? (I'm not sure they all build)

**Comment by Dekken** [comment]
> question : do we keep these 8 commits into master? (I'm not sure they all build)

I'ma squash

**Pull request merged**